### ISSUE TITLE
Firefox fails to build with some node error

### ISSUE THREAD

**Issue by LoganDark** [ehawkvu/kiss-xorg#52]
After running `kiss b firefox`, even with all dependencies installed, and waiting 50 minutes (!) for the build to get remotely close to the end, I get this weird error:

```
47:16.39 b"/root/.cache/kiss/proc/7389/build/firefox/devtools/client/shared/build/build.js:31\n    throw new Error(`\n    ^\n\nError: \n========================\nNODE COMPILATION ERROR!\n\nFile:   /root/.cache/kiss/proc/7389/build/firefox/devtools/client/debugger/src/components/Editor/Preview/ExceptionPopup.js\nStack:\n\nTypeError: unknown: Cannot read properties of null (reading '2')\n    at e.StringLiteral (/root/.cache/kiss/proc/7389/build/firefox/devtools/client/shared/build/build-debugger.js:159:62)\n    at n (/root/.cache/kiss/proc/7389/build/firefox/devtools/client/shared/build/babel.js:1:706249)\n    at e._call (/root/.cache/kiss/proc/7389/build/firefox/devtools/client/shared/build/babel.js:1:684196)\n    at e.call (/root/.cache/kiss/proc/7389/build/firefox/devtools/client/shared/build/babel.js:1:683913)\n    at e.visit (/root/.cache/kiss/proc/7389/build/firefox/devtools/client/shared/build/babel.js:1:684876)\n    at t.visitQueue (/root/.cache/kiss/proc/7389/build/firefox/devtools/client/shared/build/babel.js:1:703440)\n    at t.visitMultiple (/root/.cache/kiss/proc/7389/build/firefox/devtools/client/shared/build/babel.js:1:702943)\n    at t.visit (/root/.cache/kiss/proc/7389/build/firefox/devtools/client/shared/build/babel.js:1:703831)\n    at sA.node (/root/.cache/kiss/proc/7389/build/firefox/devtools/client/shared/build/babel.js:1:794691)\n    at e.visit (/root/.cache/kiss/proc/7389/build/firefox/devtools/client/shared/build/babel.js:1:684983)\n\n========================\n\n    at transform (/root/.cache/kiss/proc/7389/build/firefox/devtools/client/shared/build/build.js:31:11)\n    at Object.<anonymous> (/root/.cache/kiss/proc/7389/build/firefox/devtools/client/shared/build/build.js:70:16)\n    at Module._compile (node:internal/modules/cjs/loader:1112:14)\n    at Module._extensions..js (node:internal/modules/cjs/loader:1166:10)\n    at Module.load (node:internal/modules/cjs/loader:988:32)\n    at Module._load (node:internal/modules/cjs/loader:834:12)\n    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)\n    at node:internal/main/run_main_module:17:47\n\nNode.js v18.5.0\n"
47:16.41 make[3]: *** [backend.mk:1025: devtools/client/debugger/src/components/Editor/Preview/.deps/node.stub.stub] Error 1
47:16.41 make[2]: *** [/root/.cache/kiss/proc/7389/build/firefox/config/recurse.mk:34: misc] Error 2
47:16.41 make[1]: *** [/root/.cache/kiss/proc/7389/build/firefox/config/rules.mk:352: default] Error 2
47:16.41 make: *** [client.mk:63: build] Error 2
47:16.43 0 compiler warnings present.
 Config object not found by mach.
Configure complete!
Be sure to run |mach build| to pick up any changes
-> firefox Build failed
```

Wat.

Here is the full log file (renamed for GitHub):

[firefox-2022-00-00-00:00-7389.log](https://github.com/ehawkvu/kiss-xorg/files/9121048/firefox-2022-00-00-00.00-7389.log)

Running GKISS

**Comment by ehawkvu** [comment]
I'll try doing a build soon just to see if I can encounter it. @ioraff have you seen this error before?

**Comment by ioraff** [comment]
Strange. Just ran a successful build.

**Comment by LoganDark** [comment]
Gotta love these errors from high-level interpreted languages that somehow still rely on low level system details

**Comment by LoganDark** [comment]
CHROMIUM ALSO FAILS LMAO

```
[5393/50312] ACTION //third_party/devtools-frontend/src/front_end/legacy_test_runner/sources_test_runner:gen(//build/toolchain/linux/unbundle:default)
ninja: job failed: python3 ../../third_party/node/node.py ../../third_party/devtools-frontend/src/scripts/build/ninja/copy-files.js ../../third_party/devtools-frontend/src/front_end/legacy_test_runner/sources_test_runner gen/third_party/devtools-frontend/src/front_end/legacy_test_runner/sources_test_runner AutocompleteTestRunner.js,DebuggerTestRunner.js,EditorTestRunner.js,LiveEditTestRunner.js,SearchTestRunner.js,SourcesTestRunner.js,sources_test_runner.js
Traceback (most recent call last):
  File "/home/ld/.cache/kiss/proc/28698/build/chromium/out/Release/../../third_party/node/node.py", line 39, in <module>
    RunNode(sys.argv[1:])
  File "/home/ld/.cache/kiss/proc/28698/build/chromium/out/Release/../../third_party/node/node.py", line 34, in RunNode
    raise RuntimeError('Command \'%s\' failed\n%s' % (' '.join(cmd), err))
RuntimeError: Command '/home/ld/.cache/kiss/proc/28698/build/chromium/out/Release/../../third_party/node/linux/node-linux-x64/bin/node ../../third_party/devtools-frontend/src/scripts/build/ninja/copy-files.js ../../third_party/devtools-frontend/src/front_end/legacy_test_runner/sources_test_runner gen/third_party/devtools-frontend/src/front_end/legacy_test_runner/sources_test_runner AutocompleteTestRunner.js,DebuggerTestRunner.js,EditorTestRunner.js,LiveEditTestRunner.js,SearchTestRunner.js,SourcesTestRunner.js,sources_test_runner.js' failed
node:internal/modules/cjs/loader:949
  throw err;
  ^

Error: Cannot find module '/third_party/devtools-frontend/src/scripts/build/ninja/copy-files.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:946:15)
    at Module._load (node:internal/modules/cjs/loader:787:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
    at node:internal/main/run_main_module:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v18.5.0

ninja: subcommand failed
```

[chromium-2022-00-00-00:00-28698.log](https://github.com/ehawkvu/kiss-xorg/files/9125529/chromium-2022-00-00-00.00-28698.log)

What the fuck is wrong with these chains of misused high-level interpreted programming languages lmaoo

I'm so confused and sad that this is the state of build systems

**Comment by LoganDark** [comment]
Looks like this was an issue with how I compiled node.js. I used the flag `-Ofast` which implied a lot of aggressive nonstandard optimizations. I've since switched to `-O3` which, while still not exactly recommended, is allowing me to compile Chromium far past the step 5393 that failed. I'm now at 8079 with no sign of issue so far.

This is surprising, even despite what `-Ofast` does. I never expected it to cause failures in high-level scripting languages.

I am pretty sure this should extend to Firefox as well. I'll not waste my Chromium build, and wait for it to finish before trying Firefox, but once that's done, I'll try building Firefox and close this issue if it succeeds.

I'll need to look into rebuilding my system again (not just node.js), since most of it (including glibc, some other interpreters, and my entire X11 stack) was already compiled with `-Ofast`. Only one way to find out if `kiss` can handle rebuilding a library that other packages depend on.
